### PR TITLE
develop guide: add sample applications - sequelize

### PR DESCRIPTION
**Pull request by shczhen**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC-tidb-cloud.md [hunk 1]
```diff
@@ -41,6 +41,7 @@
       - [Prisma](/develop/dev-guide-sample-application-nodejs-prisma.md)
       - [Next.js](/develop/dev-guide-sample-application-nextjs.md)
       - [AWS Lambda](/develop/dev-guide-sample-application-aws-lambda.md)
+      - [Sequelize](/develop/dev-guide-sample-application-nodejs-sequelize.md)
```
Comment (Mini256) [inline TOC-tidb-cloud.md:44]:
```suggestion
      - [Sequelize](/develop/dev-guide-sample-application-nodejs-sequelize.md)
      - [Next.js](/develop/dev-guide-sample-application-nextjs.md)
      - [AWS Lambda](/develop/dev-guide-sample-application-aws-lambda.md)
```

File: TOC.md [hunk 2]
```diff
@@ -42,6 +42,7 @@
       - [Prisma](/develop/dev-guide-sample-application-nodejs-prisma.md)
       - [Next.js](/develop/dev-guide-sample-application-nextjs.md)
       - [AWS Lambda](/develop/dev-guide-sample-application-aws-lambda.md)
+      - [Sequelize](/develop/dev-guide-sample-application-nodejs-sequelize.md)
```
Comment (Mini256) [inline TOC.md:45]:
```suggestion
      - [Sequelize](/develop/dev-guide-sample-application-nodejs-sequelize.md)
      - [Next.js](/develop/dev-guide-sample-application-nextjs.md)
      - [AWS Lambda](/develop/dev-guide-sample-application-aws-lambda.md)
```

File: TOC-tidb-cloud.md (hunk 1 above)
Comment (sykp241095) [reply TOC-tidb-cloud.md:44]: What's the difference. @Mini256

Comment (Mini256) [reply TOC-tidb-cloud.md:44]: Put ORM together

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+4. Run the following command to copy `.env.example` and rename it to `.env`:
+
+    ```shell
+    cp .env.example .env
+    ```
+
+5. Edit the `.env` file, set up the environment variable `DATABASE_URL` as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
+
+    ```shell
+    TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:137]:
```suggestion
    TIDB_HOST='{host}'
```

there is no concept of region

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+    TIDB_PORT='4000'
+    TIDB_USER='{prefix}.root'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:139]:
```suggestion
    TIDB_USER='{user}'
```

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+5. Run the following command to copy `.env.example` and rename it to `.env`:
+
+    ```shell
+    cp .env.example .env
+    ```
+
+6. Edit the `.env` file, set up the environment variable `DATABASE_URL` as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
+  
+    ```shell
+    TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:107]:
```suggestion
    TIDB_HOST='{host}'
```

there is no concept of region

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+    TIDB_PORT='4000'
+    TIDB_USER='{prefix}.root'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:109]:
```suggestion
    TIDB_USER='{user}'
```

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+
+### Step 4: Run the sample app
+
+Run the following command to execute the sample code:
+
+```shell
+npm start
+```
+
+**Expected output(partial):**
```
Comment (ran-huang) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:183]: 比较长的输出能否改成 `<details><summary>` 形式

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+});
+logger.info('Read all players with coins > 300.');
+logger.info(allPlayersWithCoinsGreaterThan300.map((p) => p.toJSON()));
+```
+
+For more information, refer to [Query data](/develop/dev-guide-get-data-from-single-table.md).
+
+### Update data
+
+The following query set `700` coins and `700` goods to the `Players` with ID `6` that created in the [Insert data](#insert-data) section:
```
Comment (ran-huang) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:285]:
```suggestion
The following query set `700` coins and `700` goods to the `Players` with ID `6` that was created in the [Insert data](#insert-data) section:
```

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+await newPlayer.destroy();
+const deletedNewPlayer = await playersModel.findByPk(6);
+logger.info('Deleted the new player.');
+logger.info(deletedNewPlayer?.toJSON());
+```
+
+For more information, refer to [Delete data](/develop/dev-guide-delete-data.md).
+
+## Useful notes
+
```
Comment (ran-huang) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:311]:
```suggestion
```

File: develop/dev-guide-sample-application-nodejs-sequelize.md (hunk 7 above)
Comment (shczhen) [reply develop/dev-guide-sample-application-nodejs-sequelize.md:183]: 完整log有很多info信息没有意义。。。

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+2. Edit the `.env` file, set up the environment variable `DATABASE_URL` as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
+
+    ```shell
+    TIDB_HOST='{host}'
+    TIDB_PORT='4000'
+    TIDB_USER='root'
+    TIDB_PASSWORD='{password}'
+    TIDB_DB_NAME='test'
+    ```
```
Comment (Mini256) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:165]: Example code is forced to use SSL, which cannot run normally on TIDB Self-Hosted.

Better to have a environment variable to control it, like `TIDB_ENABLE_SSL`.

```
[18:06:59.126] INFO (app/86765): Getting sequelize instance...
[18:06:59.194] ERROR (lib/tidb/86765): Unable to connect to the database:
[18:06:59.194] ERROR (lib/tidb/86765): Server does not support secure connnection
    err: {
      "type": "ConnectionError",
      "message": "Server does not support secure connnection",
      "stack":
```

Comment (shczhen) [reply develop/dev-guide-sample-application-nodejs-sequelize.md:167]: Done.

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+
+6. Edit the `.env` file, set up the environment variable `DATABASE_URL` as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
+  
+    ```shell
+    TIDB_HOST='{host}'
+    TIDB_PORT='4000'
+    TIDB_USER='{user}'
+    TIDB_PASSWORD='{password}'
+    TIDB_DB_NAME='test'
+    TIDB_SSL_MODE='true'
```
Comment (Mini256) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:112]: Not recommended to use `SSL_ MODE` as the variable name, is not a bool value in common SSL options.

MySQL:

image

PG:

image

Comment (shczhen) [reply develop/dev-guide-sample-application-nodejs-sequelize.md:112]: Done, use `TIDB_ENABLE_SSL` instead.

Comment (Mini256) [reply develop/dev-guide-sample-application-nodejs-sequelize.md:112]: Do forget to update the step by step guide.

**Comment by sykp241095** [comment]
@shczhen has conflicts

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 12]
```diff
@@ -0,0 +1,325 @@
...
+    >
+    > If you have generated a password before, you can either use the original password or click **Reset password** to generate a new one.
+
+5. Run the following command to copy `.env.example` and rename it to `.env`:
+
+    ```shell
+    cp .env.example .env
+    ```
+
+6. Edit the `.env` file, set up the environment variable `DATABASE_URL` as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
```
Comment (Mini256) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:104]: Please check the description again.

Comment (Mini256) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:104]:
```suggestion
6. Edit the `.env` file, set up the environment variables as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
```

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 13]
```diff
@@ -0,0 +1,325 @@
...
+
+    For more details about how to obtain the connection string, refer to [TiDB Dedicated standard connection](https://docs.pingcap.com/tidbcloud/connect-via-standard-connection).
+
+4. Run the following command to copy `.env.example` and rename it to `.env`:
+
+    ```shell
+    cp .env.example .env
+    ```
+
+5. Edit the `.env` file, set up the environment variable `DATABASE_URL` as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
```
Comment (Mini256) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:135]:
```suggestion
5. Edit the `.env` file, set up the environment variables as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
```

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 14]
```diff
@@ -0,0 +1,325 @@
...
+
+<div label="TiDB Self-Hosted">
+
+1. Run the following command to copy `.env.example` and rename it to `.env`:
+
+    ```shell
+    cp .env.example .env
+    ```
+
+2. Edit the `.env` file, set up the environment variable `DATABASE_URL` as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
```
Comment (Mini256) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:159]:
```suggestion
2. Edit the `.env` file, set up the environment variables as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
```

File: develop/dev-guide-sample-application-nodejs-sequelize.md [hunk 15]
```diff
@@ -0,0 +1,325 @@
...
+
+5. Run the following command to copy `.env.example` and rename it to `.env`:
+
+    ```shell
+    cp .env.example .env
+    ```
+
+6. Edit the `.env` file, set up the environment variable `DATABASE_URL` as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
+  
+    ```shell
```
Comment (Mini256) [inline develop/dev-guide-sample-application-nodejs-sequelize.md:106]:
```suggestion
    ```dotenv
```

**Review by Mini256 — APPROVED** [review]

**Review by sykp241095 — APPROVED** [review]

**Comment by sykp241095** [comment]
@ran-huang PTAL~

**Review by ran-huang — APPROVED** [review]
rest LGTM

**Pull request merged**